kafka/src/main/java/io/micronaut/configuration/kafka/processor/KafkaConsumerProcessor.java

```diff
@@ -645,7 +645,18 @@ private Duration computeRetryDelay(ErrorStrategyValue errorStrategy, Duration fi
     private boolean processConsumerRecordsAsBatch(final ConsumerState consumerState,
                                                   final ExecutableMethod<?, ?> method,
                                                   final Map<Argument<?>, Object> boundArguments,
+                                                  final Optional<Argument<?>> ackArg,
```

Comment:
don't use an Optional as a method parameter. annotate the parameter with @Nullable.

Reply:
Agreed, my preference as well. I was staying consistent with the existing code (see `processConsumerRecords`). `processConsumerRecords` needs some other cleanup that I've got a separate branch for, so I will just make the change here for now and follow on with the same in another PR for `processConsumerRecords`.